### ISSUE TITLE
Dynamic cross-platform loading!

### ISSUE THREAD

**Issue by hpretila** [hpretila/llama.net#1]
Cross-platform loading is broken - right now need it working to package library for NuGet.

**Comment by werddomain** [comment]
is this issue is resolved?
I have problem working with windows ...

> llama_model_load: loading model from 'B:\ia\linux\llama.cpp\models\7B\ggml-model-f16.bin' - please wait ...
> llama_model_load: invalid model file 'B:\ia\linux\llama.cpp\models\7B\ggml-model-f16.bin' (bad magic)
> llama_init_from_file: failed to load model

Can this error be related?

**Comment by MichaelNichxls** [comment]
[whisper.net](https://github.com/sandrohanea/whisper.net) is a dotnet binding for [whisper.cpp](https://github.com/ggerganov/whisper.cpp). The creator managed to package it for NuGet. Maybe it could help you with that.

**Comment by hpretila** [comment]
@werddomain Apologies for the late response; this error is most likely associated to the models themselves. Lately LLaMA.CPP has been making significant changes with the model format, so even a version prior to the version of LLaMA used by this repo will not be compatible. See #3 for the issue on packing the quantisation executable with the library.

You would either:
* Have to quantise with the version I made - I have packed built quantize executables and scripts to complete the quantisation for some platforms (See [LLaMA.NET/runtimes/osx-arm64/](https://github.com/hpretila/llama.net/tree/main/LLaMA.NET/runtimes/osx-arm64/)).
* Or updating the llama.cpp submodule on this project to the versions that quantised your model.
